libenv/sysinfo.c

```diff
@@ -2311,6 +2317,31 @@ static int Linux_Mandriva_Version_Real(EvalContext *ctx, char *filename, char *r
 
 /******************************************************************/
 
+static int Linux_Amazon_Version(EvalContext *ctx)
+{
+    char buffer[CF_BUFSIZE];
+
+    // Amazon Linux AMI release 2016.09
+
+    if (ReadLine("/etc/system-release", buffer, sizeof(buffer)))
+    {
+       if (strstr(buffer, "Amazon"))
+       {
+           char version[CF_MAXVARSIZE], class[CF_MAXVARSIZE];
+           EvalContextClassPutHard(ctx, "amazon", "inventory,attribute_name=none,source=agent");
```

Comment:
This would be the first time derived-from= references a file (seems like its mostly for $(sys) variables). I don't have an objection to adding this but it seems like it should be added for everything. If that is the case I would be happy to open up a new pull request